tapdb/sqlc/migrations/000007_universe.up.sql

```diff
@@ -130,6 +130,14 @@ CREATE TABLE IF NOT EXISTS federation_uni_sync_config (
         (asset_id IS NULL AND group_key IS NOT NULL)
     ),
 
+    -- This field is similar to asset_id but is never NULL, and only used for
+    -- the unique constraint below.
+    populated_asset_id BLOB GENERATED ALWAYS AS (COALESCE(asset_id, 0)) STORED,
+
+    -- This field is similar to group_key but is never NULL, and only used for
+    -- the unique constraint below.
+    populated_group_key BLOB GENERATED ALWAYS AS (COALESCE(group_key, 0)) STORED,
```

Comment:
TIL about generated columns!

Reply:
IIUC, w/ sqlite `ALTER TABLE`, we can add columns, but _not_ drop prior `UNIQUE` constraints. 

Also the query needs to be updated here, since these new generated columns will be the ones that conflict.

Reply:
IIUC we don't actually need to drop the existing constraint, but add the new stricter constraint?

Good point re: query.